spec/spec_helper_acceptance.rb

```diff
@@ -164,7 +163,6 @@ def teardown_puppet_on(host)
 package { ['puppet-agent', 'puppet', 'mcollective', 'mcollective-client']: ensure => purged }
   EOS
   on host, puppet('apply', '-e', "\"#{pp}\"")
-  remove_defaults_on host
```

Comment:
I think it turned out this whole fix is only needed due to BKR-395; install_puppet_on sets the type to `foss` all on its own. I'll update the commit.